docs/sdk/src/reference_docs/transaction_extensions.rs

```diff
@@ -72,7 +72,7 @@ pub mod transaction_extensions_example {
 
 	// This doesn't actually check anything, but simply allows
 	// some arbitrary `u32` to be added to the extrinsic payload
-	#[derive(Debug, Encode, Decode, Clone, Eq, PartialEq, TypeInfo)]
+	#[derive(Debug, Encode, Decode, DecodeWithMemTracking, Clone, Eq, PartialEq, TypeInfo)]
```

Comment:
I wonder if we should write a new derive macro `Codec` which derives `Encode`, `Decode`, `DecodeWithMemTracking`?

Reply:
I don't know. I don't have strong preferences. I guess we could, just as we have the `Codec: Decode + Encode` trait. 

Also if we do this we should change `Codec` to `Codec: Decode + DecodeWithMemTracking + Encode ` for consistency and that could be breaking. So if we want to do it now maybe `Codec` should derive only `Encode` and `Decode` and in `parity-scale-codec v4` we can add `DecodeWithMemTracking` to it.

But anyway if we do it, I think we should have a separate issue. And we'll have to do it in `parity-scale-codec` then propagate it to `polkadot-sdk`. I don't know. If you have strong preferences I can create an issue in `parity-scale-codec`

Reply:
Yes I also don't know, maybe some `FullDecode` traits and derive macro?

Reply:
Yes, but then we need to replace `Decode` with `FullDecode` and I think it would be still the same amount of work. Also I saw there was a `FullEncode` which is unrelated to `DecodeWithMemTracking`. So the `FullDecode` may bring confusion.